### DISCUSSION TITLE
programmatically create slides using something like a for loop directive

### DISCUSSION THREAD

**Discussion by gureckis** [slidevjs/slidev#1572] [Ideas]
In #1568 I came up with a way to expose some static data to the slides (e.g., I might like to configure what semester it is in my class once in a config file and then refer to the value throughout the slide deck).  The next thing would be to programmatically create slides.  For example, my config file might include the names of all the students in the class and I'd like to make a slide for each of them individually.  Rather than typing all this into the MD I'd like to just v-for='student in students' or something where the student data is loaded from a .csv file or something.  I know I can do this in a custom .vue component and step through using v-click but in some cases you actually want distinct slides.  Another approach I consider was I could auto generate a students.md file prior to the build step and `src: ` it but it'd be neat if the .md syntax admitted some type of simple programmatic generation.

something like 

```
~~~~~~~~
for student in students:
---
layout: default
---

# {{ student.name}}

<img src="{{ student.photo }}" />
~~~~~~~~~
```

**Comment by JoeyGrajciar** [comment]
I guess answer to your question lies in https://sli.dev/custom/config-parser#use-case-1-compact-syntax-top-level-presentation, you could use the `transformRawLines` to do it. Probably easiest way would be to join the lines with new line, passing it to some templating framework and splitting the result back into lines.
